app/models/spree/back_in_stock_notification.rb

```diff
@@ -19,11 +19,16 @@ class Spree::BackInStockNotification < ApplicationRecord
   scope :pending, -> { where(email_sent_at: nil) }
 
   scope :in_stock, -> (stock_location) do
-    joins(variant: :stock_items)
-      .distinct
+    available_variants = joins(variant: [:stock_items, :product])
       .where(stock_location: stock_location)
-      .where(variant: { spree_stock_items: { stock_location_id: stock_location.id }})
-      .where("count_on_hand > ? OR backorderable = ?", 0, true)
+      .where(spree_stock_items: {stock_location_id: stock_location.id })
+      .where.not(spree_products: {available_on: nil})
+      .where("spree_products.available_on < ?", Time.current)
```

Comment:
Wouldn't it be better to merge in the available scope rather than duplicate its functionality inside of this extension?

```suggestion
      .merge(Spree::Product.available)
```

This has the added benefit of factoring in any custom functionality the store has added to that scope.

Reply:
@jarednorman thanks, good call. I have made that change.

Running the tests I see the latest version Solidus also uses `spree_products.discontinue_on`, which this extension will now also use.